src/SessionSetup/SessionSetupDialog.cpp

```diff
@@ -496,7 +501,20 @@ void SessionSetupDialog::ConnectToLocal() {
   SetupLocalProcessManager();
 }
 
-void SessionSetupDialog::SetupLocalProcessManager() { SetupProcessManager(local_grpc_channel_); }
+void SessionSetupDialog::SetupLocalProcessManager() {
+  SetupProcessManager(local_grpc_channel_);
+  ui_->processLauncherWidget->SetProcessManager(process_manager_.get());
+}
+
+void SessionSetupDialog::ProcessLaunched() {
+  const ErrorMessageOr<orbit_grpc_protos::ProcessInfo>& process_info_or_error =
+      ui_->processLauncherWidget->GetLastLaunchedProcessOrError();
```

Comment:
I would solve this via sending the ProcessInfo with the signal slot mechanism. 
(I think you can't send ErrorMessageOr, since its not default constructable, but just ProcessInfo should be fine). 

Then in ProcessLauncherWidget, in the error case you just set the error and dont send a signal.

Reply:
That's way cleaner, thanks for the suggestion, I'm sending a `const ProcessInfo&` now.